nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,89 @@
+class Email {
+	constructor(subject, body) {
+		this.subject = subject;
+		this.body = body;
+	}
+}
+
+class SpamDetector {
+	badWordCheck(email) {
+		const badWords = ["viagra", "offer", "free", "business proposal"];
+
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split(" ");
+
+		const badWordsInEmail = emailBodyIntoArray.filter((item) => {
+			return badWords.includes(item.toLowerCase());
+		});
+
+		if (badWordsInEmail.length !== 0) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	subjectCheck(email) {
+		const emailSubject = email.subject;
+
+		if (emailSubject.toLowerCase() === "hello") {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	upperCaseCheck(email) {
```

Comment:
I would let this function take the just text that needs to be checked. In that sense if you suddently has a txt message where you need to check for upper case you can still use this function. Now its tied up to the email, but it does not have to be. 

```suggestion
	hasMostlyUppercaseCharacters(text) {
```

Reply:
okay